### PR TITLE
Flake8 + Ebb-lint

### PR DESCRIPTION
**Pull request by moshez**
Comply, citizen!

### CODE REVIEW

**Comment by moshez** [comment]
This fixes #9

File: src/gather/api.py [hunk 1]
```diff
@@ -53,48 +97,19 @@ class Collector(object):
...
-        if effective_name in registry:
-            raise GatherCollisionError("Attempt to double register",
-                                       registry, effective_name, objct)
-        return objct
+    exactly_one = staticmethod(_exactly_one)
 
     def register(self, name=None, transform=lambda x: x):
-        """Register
+        """
+        Register
```
Comment (markrwilliams) [inline src/gather/api.py]: Register _a callable_.

File: src/gather/api.py [hunk 2]
```diff
@@ -53,48 +97,19 @@ class Collector(object):
...
 
-        :param name: optional. Name to register as (default is name of object)
-        :param transform: optional. A one-argument function. Will be called,
+        Args:
+            name (str): optional. Name to register as (default is name of object)
```
Comment (markrwilliams) [inline src/gather/api.py]: Name to register the _callable_ as.

-Or-

Name under which to register the callabe.

File: src/gather/api.py [hunk 3]
```diff
@@ -111,7 +126,8 @@ def main(args):
                 pass
         """
         def callback(scanner, inner_name, objct):
-            ("""Venusian_ callback to be called from scan
+            ("""
+            Venusian_ callback to be called from scan
```
Comment (markrwilliams) [inline src/gather/api.py]: Venusian callback, called from scan.

File: src/gather/api.py [hunk 4]
```diff
@@ -125,40 +141,46 @@ def callback(scanner, inner_name, objct):
...
         def attach(func):
             """Attach callback to be called when object is scanned"""
             venusian.attach(func, callback, depth=self.depth)
             return func
         return attach
 
     def collect(self, strategy=one_of.__func__):
-        """Collect all registered.
+        """
+        Collect all registered.
```
Comment (markrwilliams) [inline src/gather/api.py]: Collect all registered _callables_.

File: src/gather/api.py [hunk 5]
```diff
@@ -125,40 +141,46 @@ def callback(scanner, inner_name, objct):
...
 
         Returns a dictionary mapping names to registered elements.
         """
         def ignore_import_error(_unused):
-            """Ignore ImportError while collecting.
+            """
+            Ignore ImportError while collecting.
```
Comment (markrwilliams) [inline src/gather/api.py]: Ignore ImportError _during collection_.

File: src/gather/api.py [hunk 6]
```diff
@@ -176,29 +198,31 @@ def raise_if_needed(self):
...
-    """Run the correct subcommand.
-
-    :param argv: Arguments to be processed
-    :type argv: List of strings
-    :param commands: Commands (usually collected by a :code:`Collector`)
-    :type commands: Mapping of strings to functions that accept arguments
-    :param str version: Version string to display
-    :param file output: Where to write output to
+    """
+    Run the correct subcommand.
```
Comment (markrwilliams) [inline src/gather/api.py]: Run the _specified_ subcommands.

File: src/gather/api.py [hunk 7]
```diff
@@ -176,29 +198,31 @@ def raise_if_needed(self):
...
+        argv (list of str): Arguments to be processed
+        commands (mapping of str to callables): Commands (usually collected by a :code:`Collector`)
+        version (str): Version to display if :code:`--version` is asked
+        output (file): Where to write output to
     """
     if len(argv) < 1:
         argv = argv + ['help']
     if argv[0] in ('version', '--version'):
-        print("Version {}".format(version), file=output)
+        output.write("Version {}\n".format(version))
```
Comment (markrwilliams) [inline src/gather/api.py]: If you no longer use the `print` function in this module, you can remove [the future flag at the top of the file](https://github.com/moshez/gather/pull/17/files#diff-cf6c32c495bfb435cbe20041c1ae806fR24).

File: src/gather/example/main.py [hunk 8]
```diff
@@ -2,21 +2,26 @@
 
 from __future__ import print_function
```
Comment (markrwilliams) [inline src/gather/example/main.py]: You can also remove this.

File: src/gather/api.py [hunk 9]
```diff
@@ -32,18 +32,62 @@
...
 class GatherCollisionError(ValueError):
     """Two or more plugins registered for the same name."""
 
+
+def _one_of(_registry, _effective_name, objct):
+    """
+    Assign one of the possible options.
+
+    When given as a collection strategy to :code:`collect`,
+    will assign one of the options to a name in case more
```
Comment (moshez) [inline src/gather/api.py]: Done

File: src/gather/api.py (hunk 1 above)
Comment (moshez) [inline src/gather/api.py]: Done

File: src/gather/api.py (hunk 2 above)
Comment (moshez) [inline src/gather/api.py]: Done

File: src/gather/api.py (hunk 3 above)
Comment (moshez) [inline src/gather/api.py]: Done

File: src/gather/api.py (hunk 4 above)
Comment (moshez) [inline src/gather/api.py]: I'm changing this to "functions or classes". Technically classes are callable (sometimes -- it is possible to make a class non-callable with the right metaclass), but venusian cannot register arbitrary callables.

**Pull request merged**